core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1521,8 +1521,15 @@ abstract class RDD[T: ClassTag](
     }
 
     checkpointData match {
-      case Some(reliable: ReliableRDDCheckpointData[_]) => logWarning(
-        "RDD was already marked for reliable checkpointing: overriding with local checkpoint.")
+      case Some(reliable: ReliableRDDCheckpointData[_]) =>
```

Comment:
In addition to `checkpointData.isDefined && isCheckpointed`, we still need to check `isLocallyCheckpointed`, because when `checkpointData.isDefined` is true, it is possibly a local checkpointing.